packages/ihe-gateway-sdk/src/models/document-query.ts

```diff
@@ -1,26 +1,40 @@
-import { BaseRequest } from "./shared";
-
-export type Code = {
-  system: string;
-  code: string;
-};
+import {
+  BaseRequest,
+  BaseResponse,
+  DocumentReference,
+  XCAGateway,
+  XCPDPatientId,
+  Code,
+} from "./shared";
 
 export type DateRange = {
   dateFrom: string;
   dateTo: string;
 };
 
-export type DocumentQueryRequest = BaseRequest & {
-  xcaHomeCommunityId: string;
-  xcpdPatientId: {
-    id: string;
-    system: string;
-  };
+// The following are for us creating a document query request
+export type DocumentQueryRequestOutgoing = BaseRequest & {
+  cxId: string;
+  xcpdPatientId: XCPDPatientId;
   patientId?: string;
-  xcaGateway: string;
+  gateway: XCAGateway;
+  classCode?: Code[];
+  practiceSettingCode?: Code[];
+  facilityTypeCode?: Code[];
+  documentCreationDate?: DateRange;
+  serviceDate?: DateRange;
+};
+
+// The following are for us responding to a document query request
+export type DocumentQueryRequestIncoming = BaseRequest & {
+  xcpdPatientId: XCPDPatientId;
   classCode?: Code[];
   practiceSettingCode?: Code[];
   facilityTypeCode?: Code[];
   documentCreationDate?: DateRange;
   serviceDate?: DateRange;
 };
+// DocumentReference optional because the error response doesnt have it
```

Comment:
If we need to explain this here, maybe we need a better name for the type? In this case, it would be good to have a type that allows us to check if the object is of type A or B, and we can have strong type from there onwards, like:

```ts
const resp = getDocQueryResp();
if (resp.isError) {
   console.log(`Operation outcome: ${resp.operationOutcome}`);
   return;
}
console.log(`Doc ref: ${resp.documentReference}`) ; // the compiler knows `documentReference` is set
```

Could achieve this like so - also including a proposal to have a dedicated base error response so `operationOutcome` is not optional:

```ts
export type BaseResponse = {
  id: string;
  cxId: string;
  timestamp: string;
  responseTimestamp: string;
  xcpdPatientId?: { id: string; system: string };
  patientId: string;
};

export type BaseErrorResponse = BaseResponse & {
  operationOutcome: OperationOutcome;
};

export type DocumentQueryResponseOutgoing =
  | (BaseResponse & {
      isError: false;
      documentReference: DocumentReference[];
    })
  | (BaseErrorResponse & {
      isError: true;
    });
```

Reply:
I implemented this for all 3 types, though I did not included isError. I relied on the presence of operationOutcome for that.